src/Symfony/Component/Console/Tests/Output/StreamOutputTest.php

```diff
@@ -56,4 +57,17 @@ public function testDoWrite()
         rewind($output->getStream());
         $this->assertEquals('foo'.PHP_EOL, stream_get_contents($output->getStream()), '->doWrite() writes to the stream');
     }
+
+    public function testDoWriteOnFailure()
+    {
+        $resource = fopen(__DIR__.'/../Fixtures/stream_output_file.txt', 'r', false);
+        $output = new StreamOutput($resource);
+        try {
```

Comment:
why the try/catch? can't be let the original exception bubble?

Reply:
Yes. I thought it was better to report it as a failure since we're checking something is not happening. Updated.